sql/parser/normalize.go

```diff
@@ -99,6 +99,16 @@ func (v *normalizeVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
 
 	// Evaluate constant expressions.
 	if isConst(expr) {
+		// Normalize constant In and NotIn comparison expressions with DTuple datum.
+		if cmp, ok := expr.(*ComparisonExpr); ok {
+			switch cmp.Operator {
+			case In, NotIn:
+				tuple := cmp.Right.(DTuple)
```

Comment:
nit: this makes a copy which i think you don't need; does `cmp.Right.(DTuple).Normalize()` not do the right thing?